### PR TITLE
2.0 rc

### LINKED ISSUE #124: use Map() for storing config tree
#### Problem

There's No Good Reason I need to write code like this

``` js
const foo = this.config.bar && this.config.bar.baz && this.config.bar.baz.foo
```

in order to avoid the `foo of undefined` problem. And in Trails core, there's this craziness: https://github.com/trailsjs/trails/blob/master/lib/trails.js#L17-L25.
#### Solution

If I simply collapse the config tree into a Map(), I can happily say things like

``` js
this.config.get('bar.baz.foo')
```

and it'll either exist or not, but won't throw any sort of error. I also sort of wonder if lookup may be faster than traversing a handful of object references to get to where I want. 

This approach is much cleaner and I'm considering adopting this before 1.0. I need to research the number of outstanding trailpacks and see what it would cost to upgrade all of them.

### LINKED ISSUE #201: Add snyk to travisCI to search for vulnerabilities
#### Issue Description

https://snyk.io
Can be nice to set up snyk to every Trails modules to scan for vulnerabilities. We can also add a badge on Readme.

snyk search for vulnerabilities on all dependencies. Doesn't take a lot of time to set up and can prevent some security vulnerabilities.
#### Environment
- node version: all
- trails version: all
- operating system: all

### LINKED ISSUE #248: What about 2.0 LTS?
ROADMAP.md v2.0 LTS (October 2016)

### LINKED ISSUE #251: Test against node 7
#### Issue Description
All tests should be run against Node 7

### LINKED ISSUE #252: Combine trailpack-core with Trails itself
#### Issue Description
trailpack-core doesn't need to exist. Combining it with the core would reduce project management overhead, uneeded log output, and simplify some of the code in trailpack-core.

### LINKED ISSUE #256: refactor onceAny/after
#### Issue Description
- trails.onceAny should be re-formatted
- both onceAny and after should accept either a Promise or a callback

### LINKED ISSUE #259: make Error types available in the global namespace
#### Issue Description
If the application irrecoverably breaks during runtime, there's no guarantee a valid reference to "trails" or "lib" or anything is necessarily available. Following the design lead of nodejs itself, `Error` objects should be global.

### PR DESCRIPTION
**Pull request by tjwebb**
#### Description
2.0 release candidate branch

#### Issues
- [x] resolves #124 
- [x] resolves #256 
- [x] resolves #252
- [x] resolves #251 
- [x] resolves #248 
- [x] resolves #201 
- [x] resolves #259

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -161,34 +199,49 @@ module.exports = class TrailsApp extends events.EventEmitter {
...
+   * accepts a callback.
+   * @return Promise
    */
-  onceAny (events, handler) {
-    const self = this
-
-    if (!events)
-      return
-    if (!Array.isArray(events))
+  onceAny (events, handler = NOOP) {
```
Comment (jaumard) [inline index.js]: This will not work for node < 6 :( default value are comping with node 6

Comment (tjwebb) [inline index.js]: yep, tests are failing, I'll fix.

File: index.js [hunk 2]
```diff
@@ -12,26 +14,32 @@ module.exports = class TrailsApp extends events.EventEmitter {
...
   constructor (app) {
     super()
 
-    if (!process.env.NODE_ENV) {
-      process.env.NODE_ENV = 'development'
-    }
     if (!app.pkg) {
       throw new lib.Errors.PackageNotDefinedError()
     }
+    if (!app.api && !(app && app.api)) {
```
Comment (jaumard) [inline index.js]: One seems useless here right ?

**Comment by scott-wyatt** [comment]
Can we add #236 to this PR?

**Comment by scott-wyatt** [comment]
We should also look at #216 for the RC.

**Pull request merged**